src/frontend/src/components/anchorInput.ts

```diff
@@ -140,3 +149,38 @@ const inputFilter = (inputFilter: (c: string) => boolean, onBad: () => void) =>
       }
     }
   };
+
+/* A lit-html directive that performs an action when the element is added
+ * to the DOM.
+ *
+ * Note: there are no guarantees that the callback will be called as the
+ * implementation relies on internal "lit-html" behavior. No critical behavior
+ * should depend on this.
+ */
+const mount = (callback: (elem: Element) => void): DirectiveResult =>
+  ref((e: Element | undefined) => {
+    if (e !== undefined) {
+      // This works by observing the entire document for mutations, under
+      // the assumption that the first DOM mutation to happen after the element
+      // was created is inserting the element (or its parent) in the DOM. This
+      // happens in practice but may change depending on what lit-html does.
+      //
+      // Note: The reason why we only observe exactly one mutation and then
+      // disconnect is to avoid leaking the observer to keep observing the
+      // DOM if the element was created but never mounted.
+      //
+      // Note: it would be much easier to use the "DOMNodeInsertedIntoDocument"
+      // event on the element itself, but the API is deprecated and Firefox
+      // does not support it.
+
+      const observer = new MutationObserver(() => {
+        try {
+          callback(e);
```

Comment:
TODO: ensure that element was added to DOM before calling `callback`

Reply:
assuming `e` is a direct reference - not a clone - to the `input` and that the reference is preserved there is maybe two option.

1. check that the nodes are added contains the `e`. this works only if above statement is correct - i.e. `e` needs to be in the dom

```
const observer = new MutationObserver((mutations: MutationRecord[]) => {
    const addedNodes: Node[] = mutations.reduce(
      (acc: Node[], {addedNodes}: MutationRecord) => [...acc, ...Array.from(addedNodes)],
      []
    );
    
    const isElement: boolean = addedNodes.find((addedNode: Node) => addedNode.isEqualNode(removedNode)) !== undefined;
    
    // Some other nodes were added in the DOM
    if (!isElement) {
       return;
    }
    
    try {
       ...
```

2. if indeed we have a ref that is preserved, maybe even simplier

 ```
 const observer = new MutationObserver(() => {
    if (!e.isConnected) {
       // The element is not yet parsed in the DOM
       return;
    }
    
    try {
      ...
```

Reply:
Did the latter! Thanks for the suggestion